model/src/main/kotlin/utils/Extensions.kt

```diff
@@ -183,3 +187,19 @@ fun Identifier.toPurl() = "".takeIf { this == Identifier.EMPTY }
         append('@')
         append(version.percentEncode())
     }
+
+/**
+ * Return a list of [ScanResult]s where all results contains only findings from the same directory as the [project]'s
+ * definition file.
+ */
+fun List<ScanResult>.filterByProject(project: Project): List<ScanResult> {
```

Comment:
Please make it `fun Collection<ScanResult>.filterByProject.
`

Reply:
I agree with the motiviation, but the problem is that I would have to call `toList()` then in the early return in the first line which would copy all values to a new list, which seems a bit like a waste given that we currently never use any other collection type for scan results.

Reply:
I fully agree, I've just overlooked that.